app/models/reports.rb

```diff
@@ -18,4 +18,46 @@ def journal_reports hub, warehouse
 		  		.exec_query('')
 		end
 	end 
+
+	def stock_status_by_project_code hub, warehouse
+		stock_account = Account.find_by({'code': :stock})
+		if( hub.present?  && warehouse.present? )
+	  		@result = ActiveRecord::Base.connection
+										.exec_query('select p.project_code as project_code, cs.name as commodity_source, h.name as hub, w.name as warehouse, c.name as commodity, sum(pi.quantity) as balance
+													from posting_items pi
+													inner join projects p on p.id = pi.project_id
+													inner join commodity_sources cs on cs.id = p.commodity_source_id
+													inner join commodities c on c.id = pi.commodity_id
+													inner join hubs h on h.id = pi.hub_id
+													inner join warehouses w on w.id = pi.warehouse_id
+													where pi.account_id  = ' + stock_account.id.to_s  + '
+													and pi.hub_id = ' +   hub.to_s + ' 
+												    and pi.warehouse_id = ' +  warehouse.to_s + '
+													group by commodity_source, hub, warehouse, commodity, project_code')
+		elsif (hub.present?)
+			@result = ActiveRecord::Base.connection
+										.exec_query('select p.project_code as project_code, cs.name as commodity_source, h.name as hub, w.name as warehouse, c.name as commodity, sum(pi.quantity) as balance
+													from posting_items pi
+													inner join projects p on p.id = pi.project_id
+													inner join commodity_sources cs on cs.id = p.commodity_source_id
+													inner join commodities c on c.id = pi.commodity_id
+													inner join hubs h on h.id = pi.hub_id
+													inner join warehouses w on w.id = pi.warehouse_id
+													where pi.account_id  = ' + stock_account.id.to_s  + '
+													and pi.hub_id = ' +   hub.to_s + ' 
+													group by commodity_source, hub, warehouse, commodity, project_code')
```

Comment:
I think we need to remove the "Warehouse" from the group_by, as it isn't present in this case